tools/lkl/tests/test.sh

```diff
@@ -0,0 +1,174 @@
+#!/bin/bash
+
+script_dir=$(cd $(dirname ${BASH_SOURCE:-$0}); pwd)
+basedir=$(cd $script_dir/..; pwd)
+source ${script_dir}/autoconf.sh
+
+TEST_SUCCESS=0
+TEST_FAILURE=1
+TEST_SKIP=113
+TEST_TODO=114
+TEST_BAILOUT=115
+
+print_log()
+{
+    echo " log: |"
+    while read line; do
+        echo "  $line"
+    done < $1
+}
+
+export_vars()
+{
+    if [ -z "$var_file" ]; then
+        return
+    fi
+
+    for i in $@; do
+        echo "$i=${!i}" >> $var_file
+    done
+}
+
+lkl_test_run()
+{
+    log_file=$(mktemp)
+    export var_file=$(mktemp)
+
+    tid=$1 && shift && tname=$@
+
+    echo "* $tid $tname"
+
+    start=$(date '+%s%9N')
+    # run in a separate shell to avoid -e terminating us
+    $@ 2>&1 | strings >$log_file
+    exit=${PIPESTATUS[0]}
+    stop=$(date '+%s%9N')
+
+    case $exit in
+    $TEST_SUCCESS)
+        echo "ok $tid $tname"
+        ;;
+    $TEST_SKIP)
+        echo "ok $tid $tname # SKIP"
+        ;;
+    $TEST_BAILOUT)
+        echo "not ok $tid $tname"
+        echo "Bail out!"
+        ;;
+    $TEST_FAILURE|*)
+        echo "not ok $tid $tname"
+        ;;
+    esac
+
+    delta=$(((stop-start)/1000))
+
+    echo " ---"
+    echo " time_us: $delta"
+    print_log $log_file
+    echo -e " ..."
+
+    rm $log_file
+    . $var_file
+    rm $var_file
+
+    return $exit
+}
+
+lkl_test_plan()
+{
+    echo "1..$1 # $2"
+}
+
+lkl_test_exec()
+{
+    local SUDO=""
+    local WRAPPER=""
+
+    if [ "$1" = "sudo" ]; then
+        SUDO=sudo
+        shift
+    fi
+
+    local file=$1
+    shift
+
+    if [ -n "$LKL_HOST_CONFIG_NT" ]; then
+        file=$file.exe
+    fi
+
+    if file $file | grep "interpreter /system/bin/linker" ; then
+        adb push "$file" $ANDROID_WDIR
+        if [ -n "$SUDO" ]; then
+            ANDROID_USER=root
+            SUDO=""
+        fi
+        if [ -n "$ANDROID_USER" ]; then
+            SU="su $ANDROID_USER"
+        else
+            SU=""
+        fi
+        WRAPPER="adb shell $SU"
+        file=$ANDROID_WDIR/$(basename $file)
+    elif file $file | grep PE32; then
+        WRAPPER="wine"
+    elif file $file | grep ARM; then
+        WRAPPER="qemu-arm-static"
+    fi
+
+    $SUDO $WRAPPER $file $@
+}
+
+lkl_test_cmd()
+{
+    local WRAPPER=""
+
+    if [ -n "$LKL_HOST_CONFIG_ANDROID" ]; then
+        if [ "$1" = "sudo" ]; then
+            ANDROID_USER=root
+            shift
+        fi
+        if [ -n "$ANDROID_USER" ]; then
+            SU="su $ANDROID_USER"
+        else
+            SU=""
+        fi
+        WRAPPER="adb shell $SU"
+    fi
+
+    echo "$@" | $WRAPPER sh -x
+}
+
+adb_push()
```

Comment:
maybe we can remove/delete tools/lkl/tests/android-test.sh ?

Reply:
`make test` could be replaced as well

Reply:
Yes, these are removed in the last patch of the series.

Reply:
Thanks,

in tools/lkl/Makefile there is still the target `test valgrind gdb` which calls `make -C tests`.

https://github.com/lkl/linux/pull/400/files#diff-bb2bc012ca931e05448c2b62f99c55abR107

So removing tools/lkl/tests/Makefile lets those targets orphaned while removing android-test.sh is okay.

I think `make gdb/tests/valgrind` is still useful so how about keeping `tools/lkl/tests/Makefile` ?

Reply:
For gdb and valgrind, I think we should add an env var to select the proper wrapper in test.sh. That way we can run gdb/valgrind for all commands not only boot. 

As for the tests target, I guess we can add a target into the top Makefile to run tests/run.py.

I'll do these changes in the new version I'll push soon.

Reply:
thanks too.  I agree with what you're going to update.